src/content/docs/en/guides/framework-components.mdx

```diff
@@ -121,13 +121,16 @@ import Counter from '../components/Counter.svelte';
 </div>
 ```
 
-:::caution[Passing functions as props]
-You can pass a function as a prop to a framework component, but it only works during server rendering. If you try to use the function in a hydrated component (for example, as an event handler), an error will occur.
 
-This is because functions can't be _serialized_ (transferred from the server to the client) by Astro.
-:::
 
 
+Props that are passed to interactive framework components [using a `client:*` directive](/en/reference/directives-reference/#client-directives) must be [serialized](https://developer.mozilla.org/en-US/docs/Glossary/Serialization): translated into a format suitable for transfer over a network, or storage. However Astro does not serialize every type of data structure. Therefore, there are some limitations on what can be passed as props to hydrated components.
+
+The following prop types are supported: 
+plain object, `number`, `string`, `Array`, `Map`, `Set`, `RegExp`, `Date`, `BigInt`, `URL`, `Uint8Array`, `Uint16Array`, `Uint32Array`, and `Infinity`
+
+Non-supported data structures passed to components, such as functions, can only be used during the component's server rendering and cannot be used to provide interactivity. For example, passing functions to hydrated components is not supported because Astro cannot pass functions from the server in a way that make them executable on the client.
```

Comment:
````suggestion
Non-supported data structures passed to components, such as functions, can only be used during the component's server rendering and cannot be used to provide interactivity. For example, passing functions to hydrated components is not supported because Astro cannot pass functions from the server in a way that makes them executable on the client.

```